### PR TITLE
Use watch-based cache in density e2e tests.

### PR DESCRIPTION
**Pull request by fgrzadkowski**
This fixes a TODO in test, but more importantly reduces artificial load during density tests. Listing all pods every few seconds is not a representative load and should not out-weight other requests.

Ref #4521

### CODE REVIEW

File: test/e2e/util.go [hunk 1]
```diff
@@ -877,17 +869,37 @@ func DeleteRC(c *client.Client, ns, name string) error {
...
-// Convenient wrapper around listing pods supporting retries.
-func listPods(c *client.Client, namespace string, label labels.Selector, field fields.Selector) (*api.PodList, error) {
-	maxRetries := 4
-	pods, err := c.Pods(namespace).List(label, field)
-	for i := 0; i < maxRetries; i++ {
-		if err == nil {
-			return pods, nil
-		}
-		pods, err = c.Pods(namespace).List(label, field)
+type podStore struct {
```
Comment (gmarek) [inline test/e2e/util.go]: Can you comment this struct and move it to the beginning of the file?

File: test/e2e/util.go [hunk 2]
```diff
@@ -752,18 +751,13 @@ func RunRC(c *client.Client, name string, ns, image string, replicas int) error
 		unknown := 0
 		inactive := 0
 		failedContainers := 0
-		time.Sleep(10 * time.Second)
+		time.Sleep(5 * time.Second)
```
Comment (gmarek) [inline test/e2e/util.go]: Why this change? Only to make test faster?

**Comment by gmarek** [comment]
LGTM with one minor comment.

File: test/e2e/util.go (hunk 2 above)
Comment (fgrzadkowski) [inline test/e2e/util.go]: Mostly because it was inconsistent, and with watch it's very cheap.

File: test/e2e/util.go (hunk 1 above)
Comment (fgrzadkowski) [inline test/e2e/util.go]: Done.

**Comment by fgrzadkowski** [comment]
@gmarek PTAL

File: test/e2e/util.go [hunk 3]
```diff
@@ -94,6 +97,40 @@ type ContainerFailures struct {
 	restarts int
 }
 
+// Convenient wrapper around cache.Store that returns list of api.Pod instead of interface{}.
```
Comment (gmarek) [inline test/e2e/util.go]: s/cache.Store/cache. Store/

Comment (fgrzadkowski) [inline test/e2e/util.go]: "cache.Store" is a the name of a type (with package name) :)

**Comment by fgrzadkowski** [comment]
I'm going to merge it to see at least few jenkins runs during our day.
